activerecord/lib/active_record/connection_handling.rb

```diff
@@ -44,10 +41,9 @@ module ConnectionHandling
     #
     # The exceptions AdapterNotSpecified, AdapterNotFound and +ArgumentError+
     # may be returned on an error.
-    def establish_connection(config = nil)
+    def establish_connection(config = :primary)
```

Comment:
I think `self.configuration` is way too vague to configure a model. In the bowels of AR, it's fine to infer that "configuration" means "Active Record configuration", and then assume that means "[database connection] configuration".

At the model layer, though, we should be more explicit: `User.configuration` is just unclear.